src/coreclr/interpreter/compiler.cpp

```diff
@@ -3042,11 +3042,15 @@ int InterpCompiler::GenerateCode(CORINFO_METHOD_INFO* methodInfo)
                 {
                     AddIns(INTOP_NEWOBJ_VT);
                     m_pLastNewIns->data[1] = (int32_t)ALIGN_UP_TO(vtsize, INTERP_STACK_SLOT_SIZE);
+                    m_pLastNewIns->data[2] = 0;
                 }
                 else
                 {
                     AddIns(INTOP_NEWOBJ);
                     m_pLastNewIns->data[1] = GetDataItemIndex(ctorClass);
+                    m_pLastNewIns->data[2] = (m_compHnd->getArrayRank(ctorClass) > 0)
```

Comment:
Is it possible to get a rank of 0? What does it mean in that case?

Reply:
Once this is rebased onto the updated string ctors PR, it will be using a dedicated opcode so it should be less confusing.